create SBIX font

ara4n opened the pull request.

ara4n committed:
experimental branch for rasterising svgs via phantom for sbix

ara4n committed:
switch to using precomputed 72x72 pngs from twemoji to save cpu

ara4n committed:
remove redundant run method

ara4n committed:
generate sbix that compiles (but doesn't work yet)

ara4n committed:
chuck a cmap, loca, glyf and hmtx in. glyphs are still awol tho

ara4n committed:
make it work

ara4n committed:
make ligatures work

ara4n committed:
make it actually work

ara4n committed:
make the font pass validation by macOS

ara4n committed:
add OS/2 and fix up cmap so webfont works

ara4n committed:
document the sbix build process

ara4n committed:
add extra ligatures back

ara4n committed:
remove extra lig which breaks Glyphs

ara4n committed:
fix ligatures properly

ara4n committed:
upgrade to emoji 12.0.4

ara4n commented:
The only remaining problem I can see with this is that some ligatures inexplicably don't take effect - e.g.

`1F46D 1F3FB                                ; fully-qualified     # 👭🏻 women holding hands: light skin tone`

However, this is broken on both COLR & sbix, and also seems to be Chrome specific (Safari gets it right, somehow).

turt2live commented:
(should this not be contributed upstream rather than here?)

ara4n commented:
judging by how rapidly https://github.com/mozilla/twemoji-colr/issues/50 is getting merged, i'm not holding my breath for them to accept an invasive change like this for implementing sbix...

turt2live commented:
yup :|

Do we already ship this?

ara4n commented:
yup, it’s the fallback font used for old freetype browsers in riot-web

turt2live reviewed (approved):
We should merge this then - seems to work well in practice.

ara4n committed:
uncommitted sbix hints; unclear if these are needed so may need to revert them

ara4n committed:
merge upstream

ara4n committed:
update instructions

ara4n committed:
clarify readme

ara4n committed:
more readme

ara4n committed:
unbreak sbix font on chrome/macos

ara4n committed:
bump emojibase-data to 5

ara4n committed:
explicitly add all VS-16 ligatures to appease Safari

fixes https://github.com/vector-im/riot-web/issues/9895
VS-16 list taken from https://unicode.org/emoji/charts/emoji-variants.html
will need to be kept in sync with latest emoji

t3chguy committed:
Fix Node>10 support

t3chguy committed:
Update to Twemoji 13.1

t3chguy committed:
Merge pull request #2 from matrix-org/t3chguy/sbix

SimonBrandner committed:
Update `twe-svg.zip` (#4)

t3chguy committed:
Update to Jdecked Twemoji v15.0.3

t3chguy committed:
Override 1f97a and 1f979 with fixed SVGs

ara4n committed:
workaround chrome breaking VS16 in ligatures

see https://github.com/element-hq/element-web/issues/28500
and https://issues.chromium.org/issues/383824583

ara4n committed:
update to twemoji 15.1.0

ara4n committed:
upgrade to twemoji 17.0.2; fix monochrome twemoji

(https://github.com/element-hq/element-web/issues/14695)

ara4n committed:
add wavy-dash to monochromed twemoji

ara4n committed:
fix makefile

ara4n committed:
actually, don't make BLACK emoji monochrome. ideally we'd put a border on them.
